### PR TITLE
Add default constructor for VerletList

### PR DESCRIPTION
**Pull request by streeve**
This will now match LinkedCellList

### CODE REVIEW

File: core/unit_test/tstNeighborList.hpp [hunk 1]
```diff
@@ -420,8 +420,11 @@ void testVerletListFull()
     double grid_min[3] = { box_min, box_min, box_min };
     double grid_max[3] = { box_max, box_max, box_max };
     Cabana::VerletList<TEST_MEMSPACE,Cabana::FullNeighborTag,LayoutTag>
-        nlist( Cabana::slice<0>(aosoa), 0, aosoa.size(),
-               test_radius, cell_size_ratio, grid_min, grid_max );
+        nlist_full( Cabana::slice<0>(aosoa), 0, aosoa.size(),
+                    test_radius, cell_size_ratio, grid_min, grid_max );
+
+    Cabana::VerletList<TEST_MEMSPACE,Cabana::FullNeighborTag,LayoutTag>
+        nlist = nlist_full;
```
Comment (dalg24) [inline core/unit_test/tstNeighborList.hpp]: What is the point of making a copy?

File: core/src/Cabana_VerletList.hpp [hunk 2]
```diff
@@ -530,8 +530,14 @@ class VerletList
     VerletListData<MemorySpace,LayoutTag> _data;
 
     /*!
-      \brief Given a list of particle positions and a neighborhood radius calculate
-      the neighbor list.
+      \brief Default constructor.
+    */
+    VerletList()
+    {}
```
Comment (dalg24) [inline core/src/Cabana_VerletList.hpp]: Not called anywhere

File: core/unit_test/tstNeighborList.hpp (hunk 1 above)
Comment (sslattery) [reply core/unit_test/tstNeighborList.hpp]: To check the assignment operator and ensure the default function was behaving as expected.

File: core/src/Cabana_VerletList.hpp (hunk 2 above)
Comment (sslattery) [reply core/src/Cabana_VerletList.hpp]: True - we should technically update the code on line 427 to only create the neighbor list and then do the assignment on a separate line.

File: core/unit_test/tstNeighborList.hpp (hunk 1 above)
Comment (dalg24) [reply core/unit_test/tstNeighborList.hpp]: Default function?

**Pull request merged**